src/main/java/bio/terra/testrunner/common/utils/KubernetesClientUtils.java

```diff
@@ -641,15 +641,32 @@ private static long getApiPodCount(V1Deployment deployment, String componentLabe
 
   private static long getApiReadyPods(V1Deployment deployment) throws ApiException {
     String deploymentComponentLabel = deployment.getMetadata().getLabels().get(componentLabel);
-    long apiPodCount =
+    int transientPodCounts = listPods().size();
+    logger.debug(
+        String.format(
+            "%s: Found %s pod%s.",
+            deploymentComponentLabel, transientPodCounts, transientPodCounts > 1 ? "s" : ""));
+    // Enhanced the logic of pod readiness by checking readiness of ALL service containers within
+    // the pod. Checking only a partial set of containers for readiness might convey a false sense
+    // of pod readiness when in reality certain containers could take longer time to restart and
+    // become ready compared to others.
+    long readyPodCounts =
         listPods().stream()
             .filter(
                 pod ->
-                    deploymentComponentLabel.equals(
+                    pod.getMetadata().getLabels().containsKey(componentLabel)
+                        && deploymentComponentLabel.equals(
                             pod.getMetadata().getLabels().get(componentLabel))
-                        && pod.getStatus().getContainerStatuses().get(0).getReady())
+                        && pod.getStatus().getContainerStatuses() != null
+                        && !pod.getStatus().getContainerStatuses().isEmpty()
+                        && pod.getStatus().getContainerStatuses().stream()
+                            .allMatch(status -> status.getReady()))
```

Comment:
Do you need the `!isEmpty()`? I think `allMatch` will return `true` if the stream is empty anyway. ([ref](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#allMatch-java.util.function.Predicate-))

Reply:
Good catch! I meant to remove this line when I added the allMatch condition check but forgot to do so. This has been fixed.